src/writeData/subscriptions/components/ParsingForm.tsx

```diff
@@ -0,0 +1,211 @@
+              weight={FontWeight.Regular}
+              className="create-parsing-form__text"
+            >
+              Specify the format of your messages and define rules to parse it
+              into line protocol.
+            </Heading>
+            <Grid>
+              <Grid.Row>
+                <Grid.Column widthXS={Columns.Twelve}>
+                  <Heading
+                    element={HeadingElement.H3}
+                    weight={FontWeight.Bold}
+                    className="create-parsing-form__header"
+                  >
+                    Data Format
+                  </Heading>
+                  <SelectGroup
+                    shape={ButtonShape.StretchToFit}
+                    className="retention--radio"
+                  >
+                    <SelectGroup.Option
+                      name="line-protocol"
+                      id="never"
+                      testID="line-protocol--button"
+                      active={parsing === 'lineprotocol'}
+                      onClick={() => {
+                        setParsing('lineprotocol')
+                      }}
+                      value={null}
+                      titleText="None"
+                      disabled={false}
+                    >
+                      Line Protocol
+                    </SelectGroup.Option>
+                    <SelectGroup.Option
+                      name="user"
+                      id="user"
+                      testID="user--button"
+                      active={parsing === 'json'}
+                      onClick={() => {
+                        setParsing('json')
+                      }}
+                      value={null}
+                      titleText="None"
+                      disabled={false}
+                    >
+                      JSON
+                    </SelectGroup.Option>
+                    <SelectGroup.Option
+                      name="user"
+                      id="user"
+                      testID="user--button"
+                      active={parsing === 'string'}
+                      onClick={() => {
+                        setParsing('string')
+                      }}
+                      value={null}
+                      titleText="None"
+                      disabled={false}
+                    >
+                      STRING
+                    </SelectGroup.Option>
+                  </SelectGroup>
+                </Grid.Column>
+                {parsing === 'lineprotocol' && <LineProtocolForm />}
+                {parsing === 'json' && (
+                  <JsonParsingForm
+                    formContent={formContent}
+                    updateForm={updateForm}
+                  />
+                )}
+                {parsing === 'string' && (
+                  <StringParsingForm
+                    formContent={formContent}
+                    updateForm={updateForm}
+                  />
+                )}
```

Comment:
love this. This is what I was thinking in the earlier comment for credential types